File: articles/overconfidence-and-decision-failure/r/run_all_overconfidence_workflows.R
# run_all_overconfidence_workflows.R
# Run all R workflows for Overconfidence and Decision Failure.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "overconfidence_decision_failure_workflow.R",
  "calibration_review_tables.R",
  "confidence_error_reports.R",
  "interval_coverage_diagnostics.R",
  "planning_bias_tables.R",
  "overconfidence_review_summary.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R overconfidence workflows completed.")

File: articles/overconfidence-and-decision-failure/python/run_all_overconfidence_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Overconfidence and Decision Failure."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "confidence_error_diagnostics.py",
    "calibration_scoring.py",
    "interval_coverage_analysis.py",
    "planning_fallacy_model.py",
    "model_overconfidence_checks.py",
    "overconfidence_review_queue.py",
    "decision_record_exporter.py",
    "overconfidence_decision_failure_simulation.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python overconfidence workflows completed.")


if __name__ == "__main__":
    main()
